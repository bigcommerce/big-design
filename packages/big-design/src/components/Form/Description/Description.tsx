import React, { useMemo } from 'react';

import { LinkProps } from '../../Link';
import { Small, TextProps } from '../../Typography';

import { StyledLink } from './styled';

export interface FormControlDescriptionProps {
  readonly children?: React.ReactNode;
  readonly link?: FormControlDescriptionLinkProps;
}

export type FormControlDescriptionLinkProps = Pick<LinkProps, 'external' | 'href' | 'target'> & {
  text: string;
};

export const FormControlDescription: React.FC<TextProps & FormControlDescriptionProps> = ({
  className,
  style,
  link,
  ...props
}) => {
  const renderedDescriptionLink = useMemo(() => {
    if (!link) {
      return;
    }

    return (
      link && (
        <>
          {' '}
          <StyledLink {...link}>{link.text}</StyledLink>
        </>
      )
    );
  }, [link]);

  return (
    <Small {...props}>
      {props.children}
      {renderedDescriptionLink}
    </Small>
  );
};
