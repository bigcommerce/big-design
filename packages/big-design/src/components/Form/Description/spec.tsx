import { render } from '@testing-library/react';
import React from 'react';
import 'jest-styled-components';

import { FormControlDescription as Description } from './Description';

test('renders description', () => {
  const { container } = render(<Description>Foo</Description>);

  expect(container.firstChild).toMatchSnapshot();
});

test('renders description with link', () => {
  const { container } = render(
    <Description
      link={{
        text: 'more',
        href: 'f',
        target: 't',
      }}
    >
      Foo
    </Description>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
