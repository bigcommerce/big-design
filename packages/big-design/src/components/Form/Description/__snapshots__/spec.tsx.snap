// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders description 1`] = `
.c0 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c0:last-child {
  margin-bottom: 0;
}

<p
  class="c0"
>
  Foo
</p>
`;

exports[`renders description with link 1`] = `
.c0 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c0:last-child {
  margin-bottom: 0;
}

.c1 {
  -webkit-transition: all 70ms ease-out;
  transition: all 70ms ease-out;
  -webkit-transition-property: color;
  transition-property: color;
  color: #3C64F4;
  cursor: pointer;
  font-size: 1rem;
  font-weight: 400;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:active {
  color: #0024A6;
}

.c1:hover:not(:active) {
  color: #0024A6;
}

.c2 {
  font-size: 0.875rem;
}

<p
  class="c0"
>
  Foo
   
  <a
    class="c1 c2"
    href="f"
    target="t"
  >
    more
  </a>
</p>
`;
